Asg0.8.go

```diff
@@ -5,29 +5,29 @@ import (
 	"sync"
 )
 
-func isEven(n int) bool {
-	return n%2 == 0
+func isEven(n int, m *sync.Mutex) bool {
```

Comment:
instead of passing mutex as param and calling Lock here, keep it at caller go-routine. passing it here doesn't solves the original issue.